pkg/rest/rest_caller.go

```diff
@@ -141,6 +141,87 @@ func (c *RestCaller) CallReq(req *http.Request, result interface{}) error {
 	return err
 }
 
+// CallRawAndFollowRedirect is a shorthand for wrapping the response of CallRaw within a call to FollowRedirect
+func (c *RestCaller) CallRawAndFollowRedirect(req *http.Request) (*http.Response, error) {
+	return c.FollowRedirect(c.CallRaw(req))
+}
+
+// FollowRedirect takes the output of a previous call to CallReq and makes another request IF the first one
+// contains a location header and has a statusCode of 201 or 301-307. The server of the RestBaseUrl is used
+// to build the actual url
+func (c *RestCaller) FollowRedirect(res *http.Response, err error) (*http.Response, error) {
+	if err != nil {
+		return nil, err
+	}
+	location := res.Header.Get("Location")
+
+	if location != "" && (res.StatusCode == 201 || 301 <= res.StatusCode && res.StatusCode <= 307) {
+		redirectUrl := c.RestBaseUrl()
+		redirectUrl.Path = location
+		if res.Body != nil {
+			res.Body.Close() //Close body of original response
+		}
+		req, err := http.NewRequest("GET", redirectUrl.String(), nil)
+		if err != nil {
+			return nil, err
+		}
+		return c.CallRaw(req)
+	}
+	return res, nil
+}
+
+//CallStreaming makes a rest call, follows redirects if present and streams the output to the supplied output
+func (c *RestCaller) CallStreaming(method string, path string, query map[string]string, mimeType string, body io.ReadCloser, output io.Writer, raw bool, quiet bool) error {
+	// Create the request
+	url := c.CreateUrl(path, query)
+	req, err := http.NewRequest(strings.ToUpper(method), url.String(), body)
+	req.Header.Set("Content-Type", mimeType)
+
+	//Make the actual invocation
+	res, err := c.CallRawAndFollowRedirect(req)
+	if err != nil {
+		fmt.Println(output, err)
+		return err
+	}
+	defer res.Body.Close()
+
+	//Something when wrong it seems
+	if res.StatusCode < 200 || res.StatusCode >= 300 {
+		data, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			fmt.Println(output, err)
+			return err
+		}
+		errorWithBody := BuildErrorWithBody(res, data)
+		if raw {
+			fmt.Println(output, string(errorWithBody.Body()))
+		} else {
+			fmt.Println(output, errorWithBody.Error())
+		}
+		return err
+	}
+
+	if isJsonResponse(res) {
+		//We have got a json response
+		data, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			fmt.Println(output, err)
```

Comment:
```suggestion
			fmt.Fprintln(output, err)
```

Reply:
Nice catch! Fixed it in a few other places as well!